packages/dataclass-editor/src/components/dataclass/detail/DataClassDetailContent.tsx

```diff
@@ -1,22 +1,45 @@
 import { BasicField, Button, Flex, Textarea, ToggleGroup, ToggleGroupItem } from '@axonivy/ui-components';
-import { useState } from 'react';
 import { useAppContext } from '../../../context/AppContext';
-import { classType as getClassType } from '../data/dataclass-utils';
+import type { DataClass } from '../data/dataclass';
+import { classType } from '../data/dataclass-utils';
 import './DetailContent.css';
 
 export const DataClassDetailContent = () => {
-  const { dataClass } = useAppContext();
+  const { dataClass, setDataClass } = useAppContext();
 
-  const initialClassType = getClassType(dataClass);
+  const initialClassType = classType(dataClass);
 
-  const [classType, setClassType] = useState(initialClassType);
+  const variant = (value: string) => (value === initialClassType ? 'primary' : undefined);
 
-  const variant = (value: string) => (value === classType ? 'primary' : undefined);
+  const handleClassTypeChange = (classType: string) => {
+    const newDataClass = structuredClone(dataClass);
+    switch (classType) {
+      case 'DATA':
+        newDataClass.isBusinessCaseData = false;
+        newDataClass.entity = undefined;
+        break;
+      case 'BUSINESS_DATA':
+        newDataClass.isBusinessCaseData = true;
+        newDataClass.entity = undefined;
+        break;
+      case 'ENTITY':
+        newDataClass.isBusinessCaseData = false;
+        newDataClass.entity = { tableName: '' };
+        break;
+    }
```

Comment:
I think I would solve this with if's:

```suggestion
newDataClass.isBusinessCaseData = false;
newDataClass.entity = undefined;
if (classType === 'BUSINESS_DATA') {
  newDataClass.isBusinessCaseData = true;
}        
if (classType === 'ENTITY') {
  newDataClass.entity = { tableName: '' };
}
```